scripts/unreleased-change-monitor.ts

```diff
@@ -0,0 +1,124 @@
+];
+
+type Issue = Awaited<ReturnType<Octokit['rest']['search']['issuesAndPullRequests']>>['data']['items'][0];
+
+async function getExistingIssuesFor(dependencyName: string): Promise<Issue[]> {
+  const queryString = `type:issue repo:${ GITHUB_OWNER }/${ GITHUB_REPO } ${ UCMONITOR } in:title ${ dependencyName } in:title sort:updated`;
+  const response = await getOctokit().rest.search.issuesAndPullRequests({ q: queryString });
+
+  return response.data.items;
+}
+
+async function openIssue(issue: Issue): Promise<void> {
+  await getOctokit().rest.issues.update({
+    owner: GITHUB_OWNER, repo: GITHUB_REPO, issue_number: issue.number, state: 'open',
+  });
+  console.log(`Opened issue #${ issue.number }: "${ issue.title }"`);
+}
+
+async function closeIssue(issue: Issue): Promise<void> {
+  await getOctokit().rest.issues.update({
+    owner: GITHUB_OWNER, repo: GITHUB_REPO, issue_number: issue.number, state: 'closed',
+  });
+  console.log(`Closed issue #${ issue.number }: "${ issue.title }"`);
+}
+
+async function createIssue(latestReleaseTag: string, dependency: UnreleasedChangeMonitoringDependency): Promise<void> {
+  const title = `${ UCMONITOR }: ${ dependency.name } has changes since ${ latestReleaseTag }`;
+  const body = `Unreleased Change Monitor has detected changes to ${ dependency.name } since its last release, ${ latestReleaseTag }.` +
+    `\n\nThis is a reminder to release these changes so they make it into the next Rancher Desktop release.`;
+  const response = await getOctokit().rest.issues.create({
+    owner: GITHUB_OWNER, repo: GITHUB_REPO, title, body,
+  });
+  const issue = response.data;
+
+  console.log(`Created issue #${ issue.number }: "${ issue.title }"`);
+}
+
+// Creates issues in the main Rancher Desktop repo for external
+// depndencies that have changes that have not been released.
+// Also closes issues that were previously created by this script,
+// but that are no longer relevant.
+async function checkForUnreleasedChanges(): Promise<void> {
+  const dependencyStates: DependencyState[] = await Promise.all(dependencies.map(async(dependency) => {
+    const result = await dependency.hasUnreleasedChanges();
+
+    return {
+      dependency,
+      latestReleaseTag:     result.latestReleaseTag,
+      hasUnreleasedChanges: result.hasUnreleasedChanges,
+    };
+  }));
+
+  // reconcile issues with dependency states
+  await Promise.all(dependencyStates.map(async(dependencyState) => {
+    const dependency = dependencyState.dependency;
+
+    // get issues that are relevant to this specific dependency
+    const existingIssues = await getExistingIssuesFor(dependency.name);
+
+    if (dependencyState.hasUnreleasedChanges) {
+      let issueExists = false;
+
+      await Promise.all(existingIssues.map(async(existingIssue) => {
+        const issueTitleMatchesLatestReleaseTag = existingIssue.title.includes(dependencyState.latestReleaseTag);
```

Comment:
It would be best to have some sort of delimiter or stricter matching, so that we don't match "v1.0" if we previously had a "v1.0-rc1".

Reply:
Agreed; I added a space on the beginning of the string it searches for. This will prevent a match when the title has a string that has some letters followed by the tag. I also used `endsWith` instead of `includes`, which handles cases like your example, where there are letters that follow the tag.